packages/punchcard/lib/appsync/vtl.ts

```diff
@@ -0,0 +1,78 @@
+import { number as numberShape, ShapeGuards, string as stringShape } from '@punchcard/shape';
+import { Shape } from '@punchcard/shape/lib/shape';
+import { VExpression } from './expression';
+import { set, Statement } from './statement';
+import { VFloat, Visitor, VObject, VString } from './vtl-object';
+
+/**
+ * Represents a Velocity Template program.
+ */
+export type VTL<T> = Generator<Statement<VObject | void>, T>;
+
+declare module './vtl-object' {
+  namespace VObject {
+    export function of<T extends Shape>(type: T, expr: VExpression): VObject.Of<T>;
+    export function clone<T extends VObject>(t: T, expr: VExpression): T;
+  }
+}
+VObject.of = function of<T extends Shape>(type: T, expr: VExpression): VObject.Of<T> {
+  return type.visit(Visitor.defaultInstance as any, expr) as any;
+};
+VObject.clone = function clone<T extends VObject>(t: T, expr: VExpression): T {
+  return VObject.of(VObject.typeOf(t), expr) as any;
+};
+
+/**
+ * Type of an ExpressionTemplate factory.
+ *
+ * ```ts
+ * GraphQL.string`${mustBeAGraphQLType}`;
+ * ```
+ */
+export type ExpressionTemplate<T extends Shape> = <Args extends (VObject)[]>(template: TemplateStringsArray,...args: Args) => VTL<VObject.Of<T>>;
```

Comment:
Add a space after `TemplateStringsArray,`... we gotta get Prettier set up in this repo ;)

Reply:
Prove to me prettier isn't ugly, and I'm so down. :P